File: src/main/kotlin/syncronous/ThreadPerformanceTest.java
package syncronous;

import java.util.ArrayList;

import static syncronous.CvKt.SIZE;

public class ThreadPerformanceTest {

    public static void main(String[] args) {

        long start = System.currentTimeMillis();

        ArrayList<Thread> threads = new ArrayList<>(SIZE);

        for (int i = 0; i < SIZE; i++) {
            threads.add(new Thread(() -> {
                try {
                    Thread.sleep(1000L);
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
                System.out.print("+");

            }));
        }

        threads.forEach(thread -> {
            try {
                thread.join();
                thread.start();
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        });

        System.out.println("time millis: " + (System.currentTimeMillis() - start));
    }
}

File: src/main/kotlin/syncronous/ThreadPerformanceTest.kt
package syncronous

import kotlin.concurrent.thread

fun main(args: Array<String>) {

    val start = System.currentTimeMillis()

    val jobs = List(SIZE) {
        thread {
            Thread.sleep(1000L)
            print(".")
        }
    }
    jobs.forEach {
        it.join()
    }

    println()
    println("time millis: ${System.currentTimeMillis() - start}")
}